webui/react/src/utils/data.ts

```diff
@@ -40,12 +51,7 @@ export const isSyncFunction = (fn: unknown): boolean => {
   return isFunction(fn) && !isAsyncFunction(fn);
 };
 
-export const clone = (data: any, deep = true): any => {
-  if (isPrimitive(data)) return data;
-  if (isMap(data)) return new Map(data);
-  if (isSet(data)) return new Set(data);
-  return deep ? JSON.parse(JSON.stringify(data)) : { ...data };
-};
+export const clone = <T extends CloneableStructure>(data: T): any => structuredClone(data);
```

Comment:
I think we should simply replace custom `clone` to `structuredClone`.

1 If `structuredClone` accepts more types in the future, we need to manually change types in the current way
2. `clone` dont want to return `any` because it loses type information. If we call `structuredClone` directly, it doesnt happen.